mayavi/core/file_data_source.py

```diff
@@ -98,15 +133,39 @@ class FileDataSource(Source):
                      enter_set=True, auto_set=False,
                      desc='the current time step')
 
+    sync_timestep = Bool(False, desc='if all dataset timesteps are synced')
+
+    play = Bool(False, desc='if timesteps are automatically updated')
+    play_delay = Float(0.2, desc='the delay between loading files')
+    loop = Bool(False, desc='if animation is looped')
+
+    update_files = Button('Rescan files')
+
     base_file_name=Str('', desc="the base name of the file",
                        enter_set=True, auto_set=False,
                        editor=FileEditor())
 
     # A timestep view group that may be included by subclasses.
-    time_step_group = Group(Item(name='file_path', style='readonly'),
-                            Item(name='timestep',
-                                 defined_when='len(object.file_list) > 1')
-                            )
+    time_step_group = Group(
+                          Item(name='file_path', style='readonly'),
+                          Group(
+                              Item(name='timestep',
+                                   editor=RangeEditor(
+                                       low=0, high_name='_max_timestep',
+                                       mode='slider'
+                                   ),
+                              ),
+                              Item(name='sync_timestep'),
+                              HGroup(
+                                  Item(name='play'),
+                                  Item(name='play_delay',
+                                       label='Delay'),
+                                  Item(name='loop'),
+                              ),
+                              enabled_when='len(object.file_list) > 1'
```

Comment:
Ahh, I remember why I did this.  I thought that if you start with a single file and then add a new one, it will not show the UI elements with visible_when. It appears that I am wrong.  I just tested and this does work so I will use `visible_when`.